src/mobu/models/jupyter.py

```diff
@@ -1,62 +1,161 @@
+
+    This isn't actually all that useful here, but since these models are
+    copied from jupyterlabcontroller, which *does* make use of these features,
+    it's easier than messing with the models.
+    """
+
+    class Config:
+        """Pydantic configuration."""
+
+        alias_generator = to_camel_case
+        allow_population_by_field_name = True
+
+
+# Dashify is needed to turn, e.g. "latest_weekly" into the required
+# "latest-weekly" per sqr-066.  It's also handy for SpawnerEnum.
+
+
+def dashify(item: str) -> str:
+    return item.replace("_", "-")
+
+
+class SpawnerEnum(str, Enum):
+    """This will validate that the name is entirely upper case, and
+    will produce auto() values in lower case with underscores turned to
+    dashes.
+    """
 
-    RECOMMENDED = "recommended"
-    LATEST_WEEKLY = "latest-weekly"
-    BY_REFERENCE = "by-reference"
+    def _generate_next_value_(  # type: ignore
+        name, start, count, last_values
+    ) -> str:
+        if name != name.upper():
+            raise RuntimeError("Enum names must be entirely upper-case")
+        return dashify(name.lower())
 
 
-class JupyterImage(BaseModel):
-    """Represents an image to spawn as a Jupyter Lab."""
+class NubladoEnum(str, Enum):
+    """This will validate that the name is entirely upper case, and
+    will produce auto() values in lower case.  This is exactly StrEnum from
+    Python 3.11, except for the validation step."""
 
-    reference: str = Field(
+    def _generate_next_value_(  # type: ignore
+        name, start, count, last_values
+    ) -> str:
+        if name != name.upper():
+            raise RuntimeError("Enum names must be entirely upper-case")
+        return name.lower()
+
+
+class PartialImage(CamelCaseModel):
+    path: str = Field(
         ...,
-        title="Docker reference for the image",
-        example="registry.hub.docker.com/lsstsqre/sciplat-lab:w_2021_13",
+        name="path",
+        example="lighthouse.ceres/library/sketchbook:latest_daily",
+        title="Full Docker registry path for lab image",
+        description="cf. https://docs.docker.com/registry/introduction/",
     )
-
     name: str = Field(
         ...,
-        title="Human-readable name for the image",
-        example="Weekly 2021_34",
+        name="name",
+        example="Latest Daily (Daily 2077_10_23)",
+        title="Human-readable version of image tag",
     )
-
-    digest: Optional[str] = Field(
+    digest: str = Field(
         ...,
-        title="Hash of the last layer of the Docker container",
-        description="May be null if the digest isn't known",
+        name="digest",
         example=(
-            "sha256:419c4b7e14603711b25fa9e0569460a753c4b2449fe275bb5f89743b"
-            "01794a30"
+            "sha256:e693782192ecef4f7846ad2b21"
+            "b1574682e700747f94c5a256b5731331a2eec2"
         ),
+        title="unique digest of image contents",
+    )
```

Comment:
This should be optional since we don't care about it, I think. Then when the image is manually constructed for the case where we explicitly provide an image, we don't need to pass in a dummy string and can just let it default to `None`.